src/phoenix/experimental/evals/functions/classify.py

```diff
@@ -107,43 +235,42 @@ def llm_classify(
     if generation_info := model.verbose_generation_info():
         printif(verbose, generation_info)
 
-    # Wrap the loop in a try / catch so that we can still return a dataframe
-    # even if the process is interrupted
-    try:
-        for index, prompt in enumerate(
-            tqdm(prompts, bar_format=get_tqdm_progress_bar_formatter("llm_classify"))
-        ):
-            with set_verbosity(model, verbose) as verbose_model:
-                response = verbose_model(prompt, instruction=system_instruction, **model_kwargs)
-            if not use_openai_function_call:
-                if provide_explanation:
-                    unrailed_label, explanation = (
-                        eval_template.extract_label_from_explanation(response),
-                        response,
-                    )
-                    printif(
-                        verbose and unrailed_label == NOT_PARSABLE,
-                        f"- Could not parse {repr(response)}",
-                    )
-                else:
-                    unrailed_label = response
-                    explanation = None
+    async def _run_llm_classification(prompt: str) -> Tuple[str, Optional[str]]:
+        with set_verbosity(model, verbose) as verbose_model:
+            response = await verbose_model._async_generate(
+                prompt, instruction=system_instruction, **model_kwargs
+            )
+        if not use_openai_function_call:
+            if provide_explanation:
+                unrailed_label, explanation = (
+                    eval_template.extract_label_from_explanation(response),
+                    response,
+                )
+                printif(
+                    verbose and unrailed_label == NOT_PARSABLE,
+                    f"- Could not parse {repr(response)}",
+                )
             else:
-                try:
-                    function_arguments = json.loads(response, strict=False)
-                    unrailed_label = function_arguments.get(_RESPONSE)
-                    explanation = function_arguments.get(_EXPLANATION)
-                except json.JSONDecodeError:
-                    unrailed_label = response
-                    explanation = None
-            labels[index] = _snap_to_rail(unrailed_label, rails, verbose=verbose)
-            explanations[index] = explanation
-    except (Exception, KeyboardInterrupt) as e:
-        logger.error(e)
-        print(
-            "Process was interrupted. The return value will be incomplete",
-            e,
-        )
+                unrailed_label = response
+                explanation = None
+        else:
+            try:
+                function_arguments = json.loads(response, strict=False)
+                unrailed_label = function_arguments.get(_RESPONSE)
+                explanation = function_arguments.get(_EXPLANATION)
+            except json.JSONDecodeError:
+                unrailed_label = response
+                explanation = None
```

Comment:
Non-blocking: We should do verbose logging here in the case where OpenAI functions produces a non-JSON-parseable output.

This issue pre-dates this PR. We can address in a follow-up.